### PR TITLE
Build arm image

### PR DESCRIPTION
**Pull request by 6l20**
## PR description

- builds arm images on dedicated machine

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://github.com/consensys/orchestrate/blob/master/CHANGELOG.md).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -28,38 +28,56 @@ commands:
...
-            wget "https://github.com/docker/buildx/releases/download/v<< parameters.version >>/buildx-v<< parameters.version >>.linux-amd64" -O ~/.docker/cli-plugins/docker-buildx
-            chmod a+x ~/.docker/cli-plugins/docker-buildx
-      - run:
-          name: Start multiarch/qemu-user-static container
-          command: |
-            docker run --rm --privileged multiarch/qemu-user-static:<< parameters.qemu-user-static-version >> --reset -p yes
-      - run:
-          name: Create docker-multiarch builder
+          name: Build and push image
+          no_output_timeout: 90m
```
Comment (ggarri) [inline .circleci/config.yml:51]: no needed anymore

File: .circleci/config.yml [hunk 2]
```diff
@@ -178,44 +196,48 @@ jobs:
...
-              DOCKER_IMAGE_TAG="${DOCKER_IMAGE_REF}"
-            else
-              DOCKER_IMAGE_TAG="${DOCKER_IMAGE_REF}.${TARGETARCH}"
-            fi
+  build-docker-arm-image:
+    machine:
+      image: 'ubuntu-2004:202101-01'
+    resource_class: arm.large
+    parameters:
+      docker-image-platform:
```
Comment (ggarri) [inline .circleci/config.yml:213]: no neccesary anymore

File: .circleci/config.yml [hunk 3]
```diff
@@ -490,10 +525,20 @@ workflows:
...
+          filters:
+            tags:
+              only: /^v.*/
+            branches:
+              ignore: /.*/
       - wait-for-approval:
           type: approval
           requires:
             - run-staging-e2e
+          filters:
```
Comment (ggarri) [inline .circleci/config.yml:537]: `filters` are not needed because it already depends on `run-staging-e2e`

Comment (6l20) [reply .circleci/config.yml:537]: It used to go without but nothing runs if filters are not explicit from what I saw, actually it was necessary to add them. Which is weird

File: .circleci/config.yml [hunk 4]
```diff
@@ -513,3 +558,8 @@ workflows:
           docker-image-sign: true
           requires:
             - wait-for-approval
+          filters:
```
[the remark this answers is not in the document]
Comment (6l20) [reply .circleci/config.yml:561]: Same response as above

File: .circleci/config.yml (hunk 1 above)
Comment (6l20) [reply .circleci/config.yml:51]: Ok

File: .circleci/config.yml (hunk 2 above)
Comment (6l20) [reply .circleci/config.yml:213]: Ok

File: .circleci/config.yml [hunk 5]
```diff
@@ -427,7 +443,24 @@ workflows:
...
+            branches:
+              only: main
+            tags:
+              only: /^v.*/
+      - build-docker-arm-image: # Push Develop arm image
+          name: push-orchestrate-dev-arm-image-<< matrix.docker-image-platform >>
+          matrix:
+            parameters:
+              docker-image-platform: [ "linux/arm64" ]
```
[the remark this answers is not in the document]
Comment (6l20) [reply .circleci/config.yml:463]: Ok

File: .circleci/config.yml (hunk 3 above)
Comment (6l20) [reply .circleci/config.yml:537]: Normal, as discussed

File: .circleci/config.yml [hunk 6]
```diff
@@ -437,9 +460,9 @@ workflows:
           docker-password: ${CLOUDSMITH_PASS_RW}
           filters:
             branches:
-              only: main
+              only: build_arm
```
Comment (ggarri) [inline .circleci/config.yml:463]: Remember to update this

**Review by ggarri — APPROVED** [review]

File: .circleci/config.yml (hunk 6 above)
Comment (6l20) [reply .circleci/config.yml]: Yes thx

**Pull request merged**